pymodbus/transport/transport.py

```diff
@@ -593,10 +597,10 @@ def close(self) -> None:
         if self.connections:
             with suppress(KeyError):
                 del self.connections[self]
-        if self.other_modem:
-            self.other_modem.other_modem = None
+        if hasattr(self, 'other_modem'):
+            del self.other_modem.other_modem
```

Comment:
This is an example of different coding styles...removing variables is a no-go to me, but in this case your code is just as good as mine...so I have no problem approving this part.